apiserver/facades/client/action/action.go

```diff
@@ -191,6 +193,16 @@ func (a *ActionAPI) Enqueue(arg params.Actions) (params.ActionResults, error) {
 		receiver, err := tagToActionReceiver(action.Receiver)
 		if err != nil {
 			currentResult.Error = common.ServerError(err)
+
+			// TODO(externalreality): In this specific case we
+			// perform more detailed formatting when the receiver is
+			// not found (better than "id not found", that is).
+			// should we do this in all cases? Or more specifically,
+			// should this formatting be moved into the function
+			// that fetches the receiver from its tag.
+			if params.IsCodeNotFound(currentResult.Error) {
```

Comment:
This processing should occur inside the tagToActionReceiver call - the call site has the knowledge to format a nice error message saying that such and such is not found. The common.ServerError(err) processing here will just convert that NotFoundError with the nice message to a CodeNotFound without any additional processing required.